statistics/handle/ddl.go

```diff
@@ -75,7 +76,13 @@ var analyzeOptionDefault = map[ast.AnalyzeOptionType]uint64{
 func (h *Handle) updateGlobalStats(tblInfo *model.TableInfo) error {
 	// We need to merge the partition-level stats to global-stats when we drop table partition in dynamic mode.
 	tableID := tblInfo.ID
-	is := h.mu.ctx.GetInfoSchema().(infoschema.InfoSchema)
+	se, err := h.pool.Get()
+	if err != nil {
+		return err
+	}
+	sctx := se.(sessionctx.Context)
+	defer h.pool.Put(se)
```

Comment:
Maybe we can put it back just after we get `InfoSchema`?

Reply:
Done.